### PR TITLE
Check secret existence

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java [hunk 1]
```diff
@@ -61,6 +61,14 @@ public ApplicationRuntimeSecretService(ApplicationRuntimeSecretRepository applic
         return getSecret(appAddress, secretIndex, false);
     }
 
+    /**
+     * Check whether a secret exists.
+     * @return {@code true} if the secret exists in the database, {@code false} otherwise.
+     */
+    public boolean doesSecretExist(String appAddress, long secretIndex) {
```
Comment (zguesmi) [inline src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java:68]: Maybe
```suggestion
    public boolean isSecretPresent(String appAddress, long secretIndex) {
```
?

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java [hunk 2]
```diff
@@ -58,4 +58,16 @@ public ApplicationRuntimeSecretController(AuthorizationService authorizationServ
...
     }
 
+    @RequestMapping(method = RequestMethod.HEAD, path = "/{appAddress}/secrets/{secretIndex}")
+    public ResponseEntity<Void> checkApplicationRuntimeSecretExistence(@PathVariable String appAddress,
+                                                                       @PathVariable long secretIndex) {
+        boolean secretExists = applicationRuntimeSecretService.doesSecretExist(appAddress, secretIndex);
+        if (secretExists) {
+            log.info("Secret found [appAddress: {}, secretIndex: {}]", appAddress, secretIndex);
+            return ResponseEntity.noContent().build();
+        } else {
```
Comment (zguesmi) [inline src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java:68]: No need for else here.

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java (hunk 1 above)
Comment (mcornaton) [reply src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java:68]: Not sure but I think I prefer the first one. Do you find it unclear?

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java (hunk 2 above)
Comment (mcornaton) [reply src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java]: Good catch, thanks!

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java (hunk 1 above)
Comment (jeremyjams) [reply src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java:68]: I agree with Zied

Comment (mcornaton) [reply src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java]: Alright, that's fixed! Thanks :)

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java [hunk 3]
```diff
@@ -58,4 +58,16 @@ public ApplicationRuntimeSecretController(AuthorizationService authorizationServ
         return ResponseEntity.noContent().build();
     }
 
+    @RequestMapping(method = RequestMethod.HEAD, path = "/{appAddress}/secrets/{secretIndex}")
+    public ResponseEntity<Void> checkApplicationRuntimeSecretExistence(@PathVariable String appAddress,
```
Comment (jeremyjams) [inline src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java:62]:
```suggestion
    public ResponseEntity<Void> isApplicationRuntimeSecretPresent(@PathVariable String appAddress,
```

Comment (mcornaton) [reply src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java]: Thanks, done!

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java [hunk 4]
```diff
@@ -58,4 +58,16 @@ public ApplicationRuntimeSecretController(AuthorizationService authorizationServ
         return ResponseEntity.noContent().build();
     }
 
+    @RequestMapping(method = RequestMethod.HEAD, path = "/{appAddress}/secrets/{secretIndex}")
+    public ResponseEntity<Void> isApplicationRuntimeSecretPresent(@PathVariable String appAddress,
+                                                                  @PathVariable long secretIndex) {
+        boolean secretExists = applicationRuntimeSecretService.isSecretPresent(appAddress, secretIndex);
```
Comment (jeremyjams) [inline src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java:64]: boolean is useless (and method is clearly understandable so no need of tmp variable)

Comment (mcornaton) [reply src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretController.java]: Sure, thanks!

File: src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java (hunk 1 above)
Comment (zguesmi) [reply src/main/java/com/iexec/sms/secret/app/ApplicationRuntimeSecretService.java]: Thanks :+1:

**Review by zguesmi — APPROVED** [review]

**Pull request merged**